core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/SearchModelImpl.java

```diff
@@ -142,6 +146,12 @@ private void init() {
     configJson = getSearchConfigJson();
   }
 
+  @JsonIgnore
+  @Override
+  public String getLanguage() {
+    return StringUtils.isNotBlank(language) ? language : getLocale().getLanguage();
```

Comment:
Thanks. Just realized that getLocale is wrong, should be instead 

Optional.ofNullable(currentPage).map(p -> p.getLanguage(false)).orElse(Locale.getDefault());